src/core/iomgr/fd_posix.c

```diff
@@ -184,7 +184,9 @@ void grpc_fd_orphan(grpc_fd *fd, grpc_iomgr_cb_func on_done, void *user_data) {
   fd->on_done_user_data = user_data;
   shutdown(fd->fd, SHUT_RDWR);
   ref_by(fd, 1); /* remove active status, but keep referenced */
+  gpr_mu_lock(&fd->watcher_mu);
   wake_all_watchers(fd);
+  gpr_mu_unlock(&fd->watcher_mu);
```

Comment:
Wow, I dropped the ball by LGTMing this. For clarity, can we rename wake_all_watchers to wake_all_watchers_locked, making it properly analogous to maybe_wake_one_watcher_locked ? That way it is clear in the future whether it already holds the lock or not.